realm/src/androidTest/java/io/realm/DynamicRealmObjectTest.java

```diff
@@ -274,6 +373,44 @@ public void testSetObjectWrongTypeThrows() {
         }
     }
 
+    public void testSetListWrongTypeThrows() {
+        realm.beginTransaction();
```

Comment:
Yes, that is on purpose. The Realm is closed on `tearDown` so that will effectively cancel the transaction at the same time.